### ISSUE TITLE
Can't install - what are the requirements?

### ISSUE THREAD

**Issue by ziebelje** [felixrieseberg/razer-chroma#3]
Can you better explain the requirements and install procedure for this? I'd love to try this out as I have no interest in writing a native C++ application using the SDK. Thanks!

```> razer-chroma@0.2.3 install C:\Users\<username>\Desktop\test\node_modules\razer-chroma
> node-gyp rebuild

C:\Users\<username>\Desktop\test\node_modules\razer-chroma>if not defined npm_config_node_gyp (node "C:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node "" rebuild )
module.js:472
    throw err;
    ^

Error: Cannot find module 'nan'
    at Function.Module._resolveFilename (module.js:470:15)
    at Function.Module._load (module.js:418:25)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at [eval]:1:1
    at ContextifyScript.Script.runInThisContext (vm.js:26:33)
    at Object.exports.runInThisContext (vm.js:79:17)
    at Object.<anonymous> ([eval]-wrapper:6:22)
    at Module._compile (module.js:571:32)
    at Immediate.<anonymous> (bootstrap_node.js:383:29)
gyp: Call to 'node -e "require('nan')"' returned exit status 1 while in binding.gyp. while trying to load binding.gyp
gyp ERR! configure error
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\lib\configure.js:305:16)
gyp ERR! stack     at emitTwo (events.js:106:13)
gyp ERR! stack     at ChildProcess.emit (events.js:191:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:215:12)
gyp ERR! System Windows_NT 10.0.14393
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Users\<username>\Desktop\test\node_modules\razer-chroma
gyp ERR! node -v v7.4.0
gyp ERR! node-gyp -v v3.4.0
gyp ERR! not ok
npm WARN enoent ENOENT: no such file or directory, open 'C:\Users\<username>\Desktop\test\package.json'
npm WARN test No description
npm WARN test No repository field.
npm WARN test No README data
npm WARN test No license field.
npm ERR! Windows_NT 10.0.14393
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install" "razer-chroma"
npm ERR! node v7.4.0
npm ERR! npm  v4.0.5
npm ERR! code ELIFECYCLE

npm ERR! razer-chroma@0.2.3 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the razer-chroma@0.2.3 install script 'node-gyp rebuild'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the razer-chroma package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs razer-chroma
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls razer-chroma
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\<username>\Desktop\test\npm-debug.log```

**Comment by ziebelje** [comment]
I managed to get this working by doing `npm install nan`, but now it's trying to compile everything. Since I'm not a C++ developer, I don't have Visual Studio installed:

```
MSBUILD : error MSB3428: Could not load the Visual C++ component "VCBuild.exe". To fix this, 1) install the .NET Framew
ork 2.0 SDK, 2) install Microsoft Visual Studio 2005 or 3) add the location of the component to the system path if it i
s installed elsewhere.  [C:\Users\<username>\Desktop\chroma\node_modules\razer-chroma\build\binding.sln]
```


Is there any way you can build this and provide the binary in your release files? I'd love to be able to have some sort of lightweight application that I can just feed my scripts to.

**Comment by ziebelje** [comment]
I'll continue updating this... :)

@felixrieseberg, I found https://github.com/felixrieseberg/windows-build-tools which I installed. That got me going, but then I had to install razer-chroma using the flag to use the correct version of the build tools.

Finally up and running. Thanks for this project! I'll leave this for you to close - it would be helpful if you documented the procedure for installing the dependencies.
